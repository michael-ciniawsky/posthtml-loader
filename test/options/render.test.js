'use strict'

const webpack = require('../helpers/compiler')

describe('Options', () => {
  describe('render', () => {
    test('{String}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            render: 'posthtml-render'
          }
        }
      }

      return webpack('options/parser/fixture.js', config)
        .then((stats) => {
          const module = stats.toJson().modules[1]

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })

    test('{Object}', () => {
      const config = {
        loader: {
          test: /\.html$/,
          options: {
            parser: require('posthtml-render')()
          }
        }
      }

      return webpack('options/parser/fixture.js', config)
        .then((stats) => {
          const module = stats.toJson().modules[1]

          expect(module.source).toMatchSnapshot()
        })
        .catch((err) => err)
    })
  })
})
