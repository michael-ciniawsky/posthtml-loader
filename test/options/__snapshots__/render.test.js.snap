// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options render {Object} 1`] = `"throw new Error(\\"Module parse failed: Unexpected token (1:4)\\\\nYou may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders\\\\n> div Hello\\\\n| \\");"`;

exports[`Options render {String} 1`] = `"throw new Error(\\"Module parse failed: Unexpected token (1:4)\\\\nYou may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders\\\\n> div Hello\\\\n| \\");"`;
